ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh

```diff
@@ -6,98 +6,134 @@ set -o pipefail
+                     | .spec.template.spec.providerSpec.value.instanceType = \"${ADDITIONAL_WORKER_VM_TYPE}\"
+              " <<< "${MACHINE_SET}")
+;;
+*azure*)
+  echo "az version:"
+  az version
+  azure_auth_location=$CLUSTER_PROFILE_DIR/osServicePrincipal.json
+  echo "Logging in with az"
+  azure_auth_client_id=$(yq-v4 .clientId < "$azure_auth_location")
+  azure_auth_client_secret=$(yq-v4 .clientSecret < "$azure_auth_location")
+  azure_auth_tenant_id=$(yq-v4 .tenantId < "$azure_auth_location")
+  azure_subscription_id=$(yq-v4 .subscriptionId < "$azure_auth_location")
+  az login --service-principal -u "$azure_auth_client_id" -p "$azure_auth_client_secret"\
+    --tenant "$azure_auth_tenant_id" --output none
+  az account set --subscription "${azure_subscription_id}"
+  echo "Setting up the boot image for the ${ADDITIONAL_WORKER_ARCHITECTURE} workers..."
+  vhd_url=$(oc -n openshift-machine-config-operator get configmap/coreos-bootimages -oyaml | \
+    yq-v4 ".data.stream \
+         | eval(.).architectures.${ADDITIONAL_WORKER_ARCHITECTURE}.\"rhel-coreos-extensions\".\"azure-disk\".url")
+  vhd_name=$(basename "${vhd_url}")
+  infra_id=$(yq-v4 '.infraID' < "${SHARED_DIR}"/metadata.json)
+  rg_name="${infra_id}-rg"
+  sa_name=$(az storage account list -g "${rg_name}" | yq-v4 '.[] | select(.name == "cluster*").name')
+  AZURE_STORAGE_KEY=$(az storage account keys list -g "${rg_name}" --account-name "${sa_name}" --query "[0].value" -o tsv)
+  export AZURE_STORAGE_KEY
+  az storage blob copy start --account-name "${sa_name}" \
+      --destination-blob "${vhd_name}" --destination-container vhd --source-uri "$vhd_url"
+  gallery_name=$(az sig list -g "${rg_name}" | yq-v4 '.[].name')
+  image_name="${infra_id}-gen2-${ADDITIONAL_WORKER_ARCHITECTURE}"
+  storage_blob_url=$(az storage blob url --account-name "${sa_name}" --container-name vhd --name "${vhd_name}" -o tsv)
+  az sig image-definition create --resource-group "${rg_name}" --gallery-name "${gallery_name}" \
+    --gallery-image-definition "${image_name}" --publisher "RedHat" --offer "rhcos" \
+    --sku "rhcos-${ADDITIONAL_WORKER_ARCHITECTURE}" --os-type linux --hyper-v-generation V2 \
+    --architecture "$(sed 's/aarch64/Arm64/;s/x86_64/x64/' <<< "${ADDITIONAL_WORKER_ARCHITECTURE}")"
 
-wait "$!"
-ret="$?"
+  region=$(az group show --name "${rg_name}" | yq-v4 '.location')
+  for i in $(seq 1 15) max; do
+    [ "$i" == max ] && { echo "Timeout exceeded while waiting for the VHD blob copy to conclude. Failing..."; exit 3; }
+    sleep 60
+    [ X"$(az storage blob show --container-name vhd --name "${vhd_name}" --account-name "${sa_name}" \
+      -o tsv --query properties.copy.status)" == X"success" ] && break
+    echo "Waiting for the VHD blob copy to conclude... (timeout in $(( 15 - i )) minutes)"
+  done
+  echo "The VHD image is now available. Creating the image version..."
+  az sig image-version create --resource-group "${rg_name}" \
```

Comment:
still need the image version ? 
https://github.com/openshift/installer/pull/6684/files 
https://github.com/openshift/installer/pull/6684#issuecomment-1361687003 
and https://github.com/openshift/release/pull/34978/files usage of image version parameter

Reply:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/35101/rehearse-35101-periodic-ci-openshift-openshift-tests-private-release-4.13-multi-nightly-azure-ipi-baselinecaps-v413-p2-f14/1610338624946049024

we use `latest` later in the reference.